alvr/xtask/src/format.rs

```diff
@@ -1,52 +1,40 @@
 use alvr_filesystem as afs;
 use std::fs;
 use std::mem;
-use walkdir::{DirEntry, FilterEntry, WalkDir};
+use std::path::PathBuf;
+use walkdir::WalkDir;
 use xshell::{cmd, Shell};
 
-fn should_format(entry: &DirEntry) -> bool {
-    if entry.file_type().is_dir() {
-        match entry.path().file_name().unwrap().to_str().unwrap() {
-            "shared" | "include" => false,
-            _ => true,
-        }
-    } else {
-        let is_cpp_file = entry
-            .path()
-            .extension()
-            .is_some_and(|ext| matches!(ext.to_str().unwrap(), "c" | "cpp" | "h" | "hpp"));
-
-        let should_skip = entry
-            .path()
-            .strip_prefix(afs::workspace_dir().join("alvr"))
-            .unwrap()
-            .to_str()
-            .is_some_and(|name| {
-                matches!(
-                    name,
-                    "server/cpp/platform/win32/NvCodecUtils.h"
-                        | "server/cpp/platform/win32/NvEncoder.cpp"
-                        | "server/cpp/platform/win32/NvEncoder.h"
-                        | "server/cpp/platform/win32/NvEncoderD3D11.cpp"
-                        | "server/cpp/platform/win32/NvEncoderD3D11.h"
-                        | "server/cpp/alvr_server/nvEncodeAPI.h"
-                )
-            });
-
-        is_cpp_file && !should_skip
-    }
-}
-
-fn get_files() -> std::iter::Filter<
-    FilterEntry<walkdir::IntoIter, impl FnMut(&DirEntry) -> bool>,
-    impl FnMut(&Result<DirEntry, walkdir::Error>) -> bool,
-> {
+fn files_to_format_paths() -> Vec<PathBuf> {
     let cpp_dir = afs::crate_dir("server").join("cpp/");
 
     WalkDir::new(cpp_dir)
         .into_iter()
-        .filter_entry(|e| should_format(e))
-        .filter(|e| !e.as_ref().unwrap().file_type().is_dir())
+        .filter_entry(|entry| {
+            let included = entry
+                .path()
+                .extension()
+                .is_some_and(|ext| matches!(ext.to_str().unwrap(), "c" | "cpp" | "h" | "hpp"));
+
+            let excluded = matches!(
```

Comment:
I changed it to full paths cuz just matching file names without the basis of a convention doesn't seem like a great idea

Reply:
To me this seems alright. I don't think we would ever make our custom source files with those names. And this is also robust if we move the files.

Reply:
Sure, but if you do that also update and fix the git thingymabob for vendored files

Reply:
Also double-check with the names of vendored files in the client, not sure if those are unique enough

Reply:
Done. And we are only checking server folder

Reply:
> Done

Oops, I meant that they should be make filename based + wildcard too for consistency, but I'll approve and you can decide whether that would be worth it or not 

> And we are only checking server folder

For the time being, but thinking ahead a little doesn't hurt